servicetalk-opentelemetry/src/main/java/io/servicetalk/opentelemetry/HeadersPropagatorSetter.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright © 2022 Apple Inc. and the ServiceTalk project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.servicetalk.opentelemetry;
+
+import io.servicetalk.http.api.HttpHeaders;
+
+import io.opentelemetry.context.propagation.TextMapSetter;
+
+import java.util.Locale;
+import javax.annotation.Nullable;
+
+final class HeadersPropagatorSetter implements TextMapSetter<HttpHeaders> {
+
+    @Override
+    public void set(@Nullable final HttpHeaders headers, final String key, final String value) {
+        if (headers != null) {
+            headers.set(key.toLowerCase(Locale.ENGLISH), value);
```

Comment:
Wow, usually modern HTTP frameworks know that headers have to be lowercase in order to work with HTTP/2. I'm surprized that their [B3Propogator](https://github.com/open-telemetry/opentelemetry-java/blob/50408d499f85d5761d0a5ed9bf9d77d5ff01fff5/extensions/trace-propagators/src/main/java/io/opentelemetry/extension/trace/propagation/B3Propagator.java#L50-L53) doesn't follow this rule.